scripts/plotting_tools.py

```diff
@@ -0,0 +1,329 @@
+    def process(ds):
+        ds = ds[variables]
+        precip_sum = ds[['crain', 'csnow', 'cicep', 'cfrzr']].to_array().sum(dim='variable')
+        for ptype in ptypes:
+            ds[f'ML_c{ptype}'] = xr.where(precip_sum >= 1, x=ds[f"ML_c{ptype}"], y=np.nan)
+            ds[f'ML_{ptype}'] = xr.where(precip_sum >= 1, x=ds[f"ML_{ptype}"], y=np.nan)
+            ds[f'ML_{ptype}_epi'] = xr.where(precip_sum >= 1, x=ds[f"ML_{ptype}_epi"], y=np.nan)
+            ds[f'ML_{ptype}_ale'] = xr.where(precip_sum >= 1, x=ds[f"ML_{ptype}_ale"], y=np.nan)
+            ds[ptype] = ds[f'ML_c{ptype}'].where(ds[f'ML_c{ptype}'] >= 1) * ds[f'ML_{ptype}'] # mask where precip sum >= 1 and where ML_cptype >= 1
+            ds[f'c{ptype}'] = xr.where(ds[f'c{ptype}'] == 1, 1, np.nan)
+        return ds
+    def process_ensemble(ds):
+        ds = ds[variables]
+        precip_sum = ds[['crain', 'csnow', 'cicep', 'cfrzr']].to_array().sum(dim='variable')
+        for ptype in ptypes:
+            ds[f'ML_{ptype}'] = xr.where(precip_sum >= 1, x=ds[f'ML_{ptype}'], y=0)
+            ds[f'ML_{ptype}_epi'] = xr.where(precip_sum >= 1, x=ds[f"ML_{ptype}_epi"], y=0)
+            ds[f'ML_{ptype}_ale'] = xr.where(precip_sum >= 1, x=ds[f"ML_{ptype}_ale"], y=0)
+
+        ptype_hier = ['frzr', 'icep', 'snow', 'rain']
+        concat = xr.concat([ds[f'ML_{ptype}'] for ptype in ptype_hier], dim='ptype')
+        concat_tle = concat.mean("time")
+        max_idx = concat_tle.argmax(dim='ptype')
+
+        concat_hrrr = xr.concat([ds[f'c{ptype}'] for ptype in ptype_hier], dim='ptype')
+        concat_hrrr_tle = concat_hrrr.mean("time")
+        max_idx_tle = concat_hrrr_tle.argmax(dim='ptype')
+
+        for i, ptype in enumerate(ptype_hier):
+            ds[f'ML_c{ptype}'] = xr.where(max_idx == i, 1, np.nan) # set categorical values
+            #ds[f'ML_max_c{ptype}'] = xr.where(max_idx == i, concat_tle[i], np.nan)
+            ds[ptype] = ds[f'ML_c{ptype}'] * ds[f'ML_{ptype}'] # set categorical multiplied by prob
+            ds[f'max_c{ptype}'] = xr.where((max_idx_tle == i) & (concat_hrrr_tle[i] != 0), concat_hrrr_tle[i], np.nan)
+        ds = ds.mean("time").apply(lambda x: x.where(x != 0, np.nan)) # sum and set 0 to nan in one line
+        return ds
+        
+    if len(files) == 1:
+        ds = xr.open_dataset(files[0]).squeeze()
+        return process(ds)
+    else:
+        ds = xr.open_mfdataset(files, parallel=True)
+        ds = process_ensemble(ds)
+        return ds
+
+
+def plot_hrrr_ptype(ds):
+    '''plot hrrr precip categorizations'''
+    projection = ccrs.LambertConformal(central_longitude=-97.5, standard_parallels=(38.5, 38.5))
+    fig, ax = plt.subplots(figsize=(7, 7), subplot_kw={'projection': projection})
+    ax.set_extent([-108, -91, 37, 47.5], crs=ccrs.PlateCarree())
+    cmaps = ["Greens", "Blues", "Greys", "Reds"]
+    ptyped = ['rain', 'snow', 'icep', 'frzr'] # ordered in terms of importance, most important plotted last
+    lat = ds['latitude']
```

Comment:
Can you make these kwargs to the function (you can leave these values as default).